src/main/java/gift/repository/ProductRepository.java

```diff
@@ -1,32 +1,74 @@
 package gift.repository;
 
 import gift.entity.Product;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
 import org.springframework.stereotype.Repository;
 
 import java.util.*;
 
 @Repository
 public class ProductRepository {
-    private final Map<Long, Product> store = new HashMap<>();
-    private long sequence = 1L;
+    private final JdbcTemplate jdbcTemplate;
+    private final SimpleJdbcInsert insert;
+    public ProductRepository(JdbcTemplate jdbcTemplate)
+    {
+        this.jdbcTemplate = jdbcTemplate;
+        this.insert = new SimpleJdbcInsert(jdbcTemplate)
+                .usingGeneratedKeyColumns("id")
+                .withTableName("products");
+    }
 
-    public List<Product> findAll() {
-        return new ArrayList<>(store.values());
+    //상품 추가하기
+    public Product save(Product product)
+    {
+        Map<String, Object> map = new HashMap<>();
+        map.put("name", product.getName());
+        map.put("price", product.getPrice());
+        map.put("image_url", product.getImageUrl());
+        Number key= insert.executeAndReturnKey(new MapSqlParameterSource(map));
+        product.setId(key.longValue());
+        return product;
     }
 
-    public Optional<Product> findById(Long id) {
-        return Optional.ofNullable(store.get(id));
+    //조회하기 (전체, id)
+    public List<Product> findAll()
+    {
+        String sql = "select * from products";
+        return jdbcTemplate.query(sql,rowMapper());
     }
-    public Product save(Product product) {
-        product.setId(sequence++);
-        store.put(product.getId(), product);
-        return product;
+
+    public Optional<Product> findById(Long id)
+    {
+        String sql = "select * from products where id=?";
+        List<Product>res=jdbcTemplate.query(sql,rowMapper(),id);
+        return res.stream().findFirst();
```

Comment:
```suggestion
        return res.stream().findAny();
```
nitnit 이 경우 `findAny`도 사용가능합니다. 어떤 차이가 있을까요~?